realm/realm-library/src/main/java/io/realm/BaseRealm.java

```diff
@@ -70,7 +72,10 @@
     // thread pool for all async operations (Query & transaction)
     static final RealmThreadPoolExecutor asyncQueryExecutor = RealmThreadPoolExecutor.getInstance();
 
-    protected final List<WeakReference<RealmChangeListener>> changeListeners =
+    protected final Set<RealmChangeListener> changeListeners =
```

Comment:
This will make it undefined in which order the listeners are called, but prevent people registering the same listener twice. Would it make more sense to store them in a list instead, then the callback sequence is a bit more predictable (nice for debugging) at the expense of a little less security.